backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/predicate/impl/BooleanFieldPredicateBuilderFactory.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.lucene.types.predicate.impl;
+
+import org.hibernate.search.backend.lucene.search.impl.LuceneSearchContext;
+import org.hibernate.search.backend.lucene.types.converter.impl.BooleanFieldConverter;
+
+public final class BooleanFieldPredicateBuilderFactory
```

Comment:
I think you can remove this class entirely: just change `IntegerFieldPredicateBuilderFactory` by replacing `StandardFieldConverter<Integer>` with `LuceneFieldConverter<?, Integer>`, and you will be able to use it with your `BooleanFieldConverter`.